apps/meteor/tests/e2e/translations.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { Users } from './fixtures/userStates';
+import { setSettingValueById } from './utils/setSettingValueById';
+import { setUserPreferences } from './utils/setUserPreferences';
+import { test, expect } from './utils/test';
+
+test.use({ storageState: Users.admin.state });
+
+test.describe('Translations', () => {
+    test.beforeAll(async ({ api }) => {
+        expect((await setSettingValueById(api, 'Language', 'en')).status()).toBe(200);
+        expect((await setSettingValueById(api, 'Site_Name', 'Rocket.Chat')).status()).toBe(200);
+	});
+
+    test.afterAll(async ({ api }) => {
+        expect((await setUserPreferences(api, { language: '' })).status()).toBe(200);
+        expect((await setSettingValueById(api, 'Language', 'en')).status()).toBe(200);
+	});
+
+    test.beforeEach(async ({ api, page }) => {
+        expect((await setUserPreferences(api, { language: '' })).status()).toBe(200);
+        await page.goto('/home');
+	})
+
+    test('expect to respect user preference', async ({ page, api }) => {
+        await expect(page.locator('h2')).toHaveText('Welcome to Rocket.Chat');
+
+        expect((await setUserPreferences(api, { language: 'pt-BR' })).status()).toBe(200);
+
+        await expect(page.locator('h2')).toHaveText('Bem-vindo ao Rocket.Chat');
+    });
+
+    test.describe('Browser language', () => {
+        test.use({ locale: 'pt-BR' })
+        test('expect to respect browser language', async ({ page }) => {
+            // await expect(page.locator('html')).toHaveAttribute('lang', 'pt-BR');
+            await expect(page.locator('h2')).toHaveText('Bem-vindo ao Rocket.Chat');
+        });
+    })
+
+    test.describe.skip('Server language', () => {
+        test.use({ locale: '' })
+        test('expect to respect server language', async ({ page, api }) => {
```

Comment:
Can we add 2 (more) test cases to ensure the page is not changing the language when the user has a language preference and another one for when they have a language in their browser too?

Asking that because I remember the mentioned bug happened because the whole page was translated whenever the user changed the server language.

Reply:
I can add the tests, but that was not a "bug" since the code was specifically implemented to change the languages when switching the setting (and only happens through the UI). I removed that bit of code. Also, I had tried to add tests for the server language, but I cannot test the server language since playwright's browser will always have a locale set, and there's no way to change that.

I actually removed this one in the latest commits.